README.md

```diff
@@ -504,6 +504,15 @@ golang:
    # SYFT_GOLANG_LOCAL_MOD_CACHE_DIR env var
    local-mod-cache-dir: ""
 
+   # search for go package licences by retrieving the package from a network proxy
+   # SYFT_GOLANG_SEARCH_REMOTE_LICENSES env var
+   search-remote-licenses: false
+
+   # remote proxy to use when retrieving go packages from the network,
+   # if unset this defaults to $GOPROXY followed by https://proxy.golang.org
```

Comment:
We may want to support the `GOPRIVATE` env var: https://github.com/goproxyio/goproxy.io/blob/master/content/docs/GOPRIVATE-env.md

Reply:
I think we should, but I would leave it for a later iteration.

Reply:
There are a few interactions with GOPROXY, GONOPROXY, and GOPRIVATE (https://go.dev/ref/mod#environment-variables).

1. GOPROXY not set or set to `direct` , which means download from the source repo
2. GOPROXY set to `off` which would disallow network access (effectively disabling this feature)
3. GOPROXY could have a `file:///` prefix which means to search the filesystem
4. GOPROXY could have multiple values specified in a comma separated list, though there is special processing where if `direct` in the list is found, then no further sources are processed
5. GOPRIVATE can select certain modules based off of a glob prefix such that those sources are always considered as `direct` and the remaining modules follow the standard GOPROXY value (also accepts comma delimited list)
6. GONOPROXY (note the "no") defaults to GOPRIVATE if not set, but can be explicitly overridden and even set to `none` (I think) to ensure everything is accessed via GOPROXY configuration.

I think just having `remote-proxy` configuration could lead to surprising results for those that are already using these configurations.

I agree with @kzantow , since we're keying off of the GO* configuration, we should try to replicate the same behavior. Even if we don't support the same functionality we should at least detect when we're configured in a way that we cannot support and output at least warnings to the log (if not error out entirely).

Reply:
This has me thinking that maybe we should just use the env vars? Meaning the setting in the config file (or env var `SYFT_GOLANG_GOPROXY` or `SYFT_GOLANG_PROXY`) really should be an _enabler_, rather than the actual value:

* `SYFT_GOLANG_GOPROXY` set to `""`? We do not get from the Internet
* `SYFT_GOLANG_GOPROXY` set to `true`? We get from the Internet, following the official rules for the various env vars as shown above.

Needless to say, all of the logic in the original `go` cmd is in `private/` [here](https://github.com/golang/go/blob/06264b740e3bfe619f5e90359d8f0d521bd47806/src/cmd/go/internal/modfetch/repo.go#L228) and similar spaces.

Reply:
I added a separate commit that does this (rather than modifying the same commit, so it will be easier to roll back).

The part I left out is the `"direct"`, as that is a messy thing to implement. [This says](https://go.dev/ref/mod#vcs):

> The go command may download module source code and metadata directly from a version control repository. 
> Downloading a module from a [proxy](https://go.dev/ref/mod#communicating-with-proxies) is usually faster, but 
> connecting directly to a repository is necessary if a proxy is not available or if a module’s repository is not accessible 
> to a proxy (frequently true for private repositories). Git, Subversion, Mercurial, Bazaar, and Fossil are supported. A 
> version control tool must be installed in a directory in PATH in order for the go command to use it.

That is going to be some ugly mess to build in and support, whether natively or using external commands (which is what `go` uses). You cannot even rely on them being there, as `syft` very well could be running (likely is) outside of the build environment.

Suggestions?